topics/basics/plugin_signing.md

```diff
@@ -0,0 +1,164 @@
+
+Plugin Signing is a mechanism introduced in the 2021.2 release cycle to increase security in JetBrains Marketplace and all of our IntelliJ-based IDEs.
+
+The Marketplace signing is designed to ensure that plugins are not modified over the course of the publishing and delivery pipeline.
+If the plugin is not signed by the author or if a certificate has been revoked, a warning dialog will appear in the IDE during installation.
+
+On our side, we will check if the public part of a key is present, and we will verify the signature. This is similar to the [Google upload key](https://developer.android.com/studio/publish/app-signing#generate-key) mechanism.
+
+## How Signing Works
+
+To be sure a file has not been modified, the file will be signed twice – once by the plugin author and once by JetBrains Marketplace.
+
+The plugin author's sign-verify process is as follows:
+
+- A user generates a key pair and uploads the public part to JetBrains Hub.
+- A build tool signs a plugin file during the assembly process.
+- The user uploads the plugin file to JetBrains Marketplace.
+- JetBrains Marketplace checks if the public key is present in the JetBrains Hub user profile.
+- JetBrains Marketplace verifies the signature.
+- The JetBrains sign-verify process is as follows:
+
+JetBrains CA is used as the source of truth here.
+Its public part will be added to the product Java TrustStore, while the private part will be used only once to generate an intermediate certificate.
+The private key of JetBrains CA is super-secret; in fact, we've already said too much.
+
+- The intermediate certificate issues a certificate that will be used to sign plugins. 
+  This way, it will be possible to re-generate this certificate without access to JetBrains CA's super-secret private key.
+  The private key of the intermediate certificate is issued and kept in the AWS Certificate Manager, and no application has access to it; people's access is also limited. 
+  So now we have an AWS-based Intermediate CA. 
+  The public part of the intermediate certificate will be added to the plugin file together with the signing certificate.
+- The certificate used to sign plugins is stored securely, too.
+  We used the AWS Key Management Service (KMS) to generate a private key, so it can never be leaked.
+  Then we prepared a certificate request (CSR) using the AWS KMS.
+  Then the CSR was signed by the Intermediate CA.
+  JetBrains Marketplace uses AWS KMS as a signature provider to sign plugin files.
+
+## Signing Methods
+
+To provide a suitable method for plugin signing, we have introduced the [Marketplace ZIP Signer](https://github.com/JetBrains/marketplace-zip-signer) library.
+It can be executed using the `signPlugin` task provided by the [Gradle IntelliJ Plugin](https://github.com/JetBrains/gradle-intellij-plugin) if your project is Gradle-based or standalone CLI.
```

Comment:
why do we need CLI tool? when using DevKit setup? other reasons?

Reply:
"or standalone CLI" --> "Alternatively, it can be used stand alone (link to paragraph)"